pkg/agent/controller/egress/ip_assign_advertise_linux.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egress
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/util"
+	"antrea.io/antrea/pkg/agent/util/arping"
+)
+
+type nodeEgressAssigner struct {
```

Comment:
The file name and the struct name don't match. The struct and interface seems not useful for now. But I think you may need the struct to be responsible for cleaning up stale EgressIPs that were configured by the agent but the corresponding Egress was deleted when the agent was down.

Reply:
> But I think you may need the struct to be responsible for cleaning up stale EgressIPs that were configured by the agent but the corresponding Egress was deleted when the agent was down.


Means unasigning egressIP after egress is deleted? How about implement it in uninstallEgress? I have not tested this situation
```
func (c *EgressController) uninstallEgress(egressName string, eState *egressState) error {
	// Uninstall all of its Pod flows.
	if err := c.uninstallPodFlows(egressName, eState, eState.ofPorts, eState.pods); err != nil {
		return err
	}
	// Release the EgressIP's mark if the Egress is the last one referring to it.
	if err := c.unrealizeEgressIP(egressName, eState.egressIP); err != nil {
		return err
	}
	if c.localIPDetector.IsLocalIP(eState.egressIP) {
		if err := c.ipAssigner.UnassignEgressIP(eState.egressIP); err != nil {
			return err
		}
	}
	// Remove the Egress's state.
	c.deleteEgressState(egressName)
	return nil
}
```

Reply:
uninstallEgress will only be called when egress is deleted when agent is running. Think about the scenario that the egress is deleted when agent is restarting. Will agent know it needs to remove the egress IP whose Egress no longer exist

Reply:
done, I think it can be solved with a egressIP storage solution